sdk/appconfiguration/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/config/AppConfigurationBootstrapConfiguration.java

```diff
@@ -60,19 +69,23 @@ AppConfigurationPropertySourceLocator sourceLocator(AppConfigurationProperties p
     }
 
     /**
+     * @param clientProperties AzureKeyVaultSecretProvider client properties
      * @throws IllegalArgumentException if both KeyVaultClientProvider and KeyVaultSecretProvider exist.
      */
     @Bean
-    AppConfigurationKeyVaultClientFactory keyVaultClientFactory() throws IllegalArgumentException {
+    AppConfigurationKeyVaultClientFactory keyVaultClientFactory(AzureKeyVaultSecretProperties clientProperties)
+        throws IllegalArgumentException {
         KeyVaultCredentialProvider keyVaultCredentialProvider = context
             .getBeanProvider(KeyVaultCredentialProvider.class).getIfAvailable();
         SecretClientBuilderSetup keyVaultClientProvider = context.getBeanProvider(SecretClientBuilderSetup.class)
             .getIfAvailable();
         KeyVaultSecretProvider keyVaultSecretProvider = context.getBeanProvider(KeyVaultSecretProvider.class)
             .getIfAvailable();
 
```

Comment:
What's each of these four, the `KeyVaultCredentialProvider`, `SecretClientBuilderSetup`, SecretClientBuilderFactory, and `KeyVaultSecretProvider` for?

Reply:
`KeyVaultCredentialProvider` - A Bean for providing credentials to us for connecting to any key vault references found.
`SecretClientBuilderSetup` - A Bean for modifying the client builder for a given key vault
`SecretClientBuilderFactory` - Spring Teams Factory for creating key vault clients.
`KeyVaultSecretProvider` - A Bean for resolving key vault references without an associated key vault. This could be a way to resolve a key vault reference in a dev environment without connecting to an actual Key Vault.

Reply:
Do you think we can use this `AzureServiceClientBuilderCustomizer<SecretClientBuilder>` to replace the `SecretClientBuilderSetup`, and `KeyVaultCredentialProvider`?

Reply:
No, as we allow the client and credentials to me modified based on the URI

Reply:
Since the `SecretClientBuilderSetup` provides way to set up the `credential` and `key`, let's see wether we can only keep the `SecretClientBuilderSetup` and remove the `KeyVaultSecretProvider` and `KeyVaultCredentialProvider`.